### PR TITLE
Simplified open correct file

### PR DESCRIPTION
**Pull request by harshad1**
This PR supersedes https://github.com/gsantner/markor/pull/970.

In this PR I address 3 issues:
1. Always open the correct file from shortcuts and the widget
2. Correctly handle multi windows
3. Widget new file is added to correct folder (#873)

### CODE REVIEW

File: app/src/main/res/xml/prefactions_share_into.xml [hunk 1]
```diff
@@ -40,7 +40,6 @@
         android:title="@string/calendar" />
 
     <PreferenceCategory
-        android:key="@string/pref_key__favourite_files"
```
Comment (harshad1) [inline app/src/main/res/xml/prefactions_share_into.xml:43]: Removed as android studio complained about this key being a duplicate. Don't think there is any detriment to removing it.

**Comment by harshad1** [comment]
@gsantner I will be testing this on another device later in the evening. If you could test it, too that would be great. Otherwise it is good to go

File: app/src/main/java/net/gsantner/markor/activity/DocumentOpener.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+
+/**
+ * This Activity exists solely to launch DocumentActivity with the correct intent
+ * it is necessary as widget and shortcut intents do not respect MultipleTask etc
+ */
+public class DocumentOpener extends Activity {
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/activity/DocumentOpener.java:11]: Suggestion postfix with Activity

File: app/src/main/res/xml/prefactions_share_into.xml (hunk 1 above)
Comment (gsantner) [reply app/src/main/res/xml/prefactions_share_into.xml:43]: I see, seems like copy paste error

File: app/src/main/java/net/gsantner/markor/activity/DocumentOpener.java (hunk 2 above)
Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/activity/DocumentOpener.java]: renamed pushed

**Comment by gsantner** [comment]
Just merged in master and renamed activity, please pull before changing code :D

**Comment by gsantner** [comment]
thinks did some minor changes and merged!

**Pull request merged**